### PR TITLE
Some code review fixes

### PR DESCRIPTION
**Pull request by nikitamishustin**

### CODE REVIEW

**Review by dchernykh79 — CHANGES_REQUESTED** [review]

File: cloudera_cm_config.py [hunk 1]
```diff
@@ -108,13 +118,18 @@ def main():
     cm_client.configuration.password = params['cm_password']
     cm_client.configuration.host = api_url
     api_client = cm_client.ApiClient()
-
     cm_config = CM(name=params["cm_host"], api_client=api_client)
```
[the remark this answers is not in the document]
Comment (nikitamishustin) [reply cloudera_cm_config.py:121]: Then we can just try-except all requests to the API.

File: cloudera_cm_config.py [hunk 2]
```diff
@@ -83,12 +86,19 @@ def _put_state(self):
             config_body.append(prop_dict)
         body = ApiConfigList(config_body)
         self.cm_resource_api_client.update_config(body=body)
-        self.cm_resource_api_client.refresh_parcel_repos()
+        # Refreshing is not momentary, we need to wait until refresh command will be inactive.
+        if 'parcel' in new_prop.lower():
```
[the remark this answers is not in the document]
Comment (nikitamishustin) [reply cloudera_cm_config.py:90]: > This API could be used following two scenarios.
> - User updated Cloudera Manager's local parcel repository.
> - User updated remote parcel locations.

So I think we can use something like this:
```
if new_prop.lower() in ('parcel_repo_path', 'remote_parcel_repo_urls'):
```

[the remark this answers is not in the document]
Comment (nikitamishustin) [reply cloudera_cm_config.py:90]: If `new_prop` is not about parcels - we can use a global refresh method: https://archive.cloudera.com/cm7/7.2.4/generic/jar/cm_api/swagger-html-sdk-docs/python/docs/ClustersResourceApi.html#refresh

File: cloudera_parcel.py [hunk 3]
```diff
@@ -43,55 +44,107 @@ def build_module():
...
+    @Decorators.try_cm_api
+    def _distribute(self):
+        self.parcel_api_client_instance.start_distribution_command(self.cluster_name, self.name, self.version)
+
+    @Decorators.try_cm_api
+    def _acrivate(self):
+        self.parcel_api_client_instance.activate_command(self.cluster_name, self.name, self.version)
+
+    @Decorators.try_cm_api
+    def _deacrivate(self):
```
Comment (Lozmanov) [inline cloudera_parcel.py:129]: deactivate

Comment (nikitamishustin) [reply cloudera_parcel.py]: fixed

File: cloudera_parcel.py [hunk 4]
```diff
@@ -43,55 +44,107 @@ def build_module():
...
+    def _download(self):
+        self.parcel_api_client_instance.start_download_command(self.cluster_name, self.name, self.version)
+
+    @Decorators.try_cm_api
+    def _distribute(self):
+        self.parcel_api_client_instance.start_distribution_command(self.cluster_name, self.name, self.version)
+
+    @Decorators.try_cm_api
+    def _acrivate(self):
+        self.parcel_api_client_instance.activate_command(self.cluster_name, self.name, self.version)
```
[the remark this answers is not in the document]
Comment (nikitamishustin) [reply cloudera_parcel.py:126]: fixed

**Pull request merged**